akka-http-core/src/main/scala/akka/http/engine/parsing/BodyPartParser.scala

```diff
@@ -115,9 +116,8 @@ private[http] final class BodyPartParser(defaultContentType: ContentType,
       }
       rec(offset)
     } catch {
-      case NotEnoughDataException ⇒ continue(input.takeRight(needle.length), 0)(parsePreamble)
+      case NotEnoughDataException ⇒ continue(input.takeRight(needle.length + 2), 0)(parsePreamble)
```

Comment:
all of these eta expansions perform one allocation per call, it should be way more efficient to store the function in a `val` instead